cve_bin_tool/checkers/__init__.py

```diff
@@ -284,10 +285,10 @@ def __new__(cls, name, bases, props):
 
 
 class Checker(metaclass=CheckerMetaClass):
-    CONTAINS_PATTERNS: list = []
-    VERSION_PATTERNS: list = []
-    FILENAME_PATTERNS: list = []
-    VENDOR_PRODUCT: list = []
+    CONTAINS_PATTERNS: list[str] = []
+    VERSION_PATTERNS: list[str] = []
+    FILENAME_PATTERNS: list[str] = []
```

Comment:
And now that I've looked at it a bit closer I'm unsure if this is typed properly as well. In all checker implementations these constants are just lists of strings but in `__new__` method of the metaclass above they all get converted into `list[re.Pattern]` 🤔

Reply:
Technically at this point the patterns are still `str` (theyre' typically "raw" `str` to help with backslash readability, but still technically `str` until the conversion)  It might be worth rethinking it but for now `str` is correct.